web/ui/react-app/src/pages/Services.tsx

```diff
@@ -1,15 +1,199 @@
-import React, { FC } from 'react';
+import React, { Component } from 'react';
 import { RouteComponentProps } from '@reach/router';
 import PathPrefixProps from '../PathPrefixProps';
-import { Alert } from 'reactstrap';
-
-const Services: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix }) => (
-  <>
-    <h2>Service Discovery</h2>
-    <Alert color="warning">
-      This page is still under construction. Please try it in the <a href={`${pathPrefix}/service-discovery`}>Classic UI</a>.
-    </Alert>
-  </>
-);
+
+interface ServicesState {
+  response: Record<string, any>;
+  fetchServicesErrorMessage: string;
+  targets: any;
+  labels: any;
+  showMore: boolean;
+}
+
+interface LabelState {
+  showMore: boolean;
+}
+
+interface LabelProps {
+  value: any;
+}
+
+class Services extends Component<RouteComponentProps & PathPrefixProps, ServicesState> {
+  constructor(props: RouteComponentProps & PathPrefixProps) {
+    super(props);
+
+    this.state = {
+      response: {},
+      fetchServicesErrorMessage: '',
+      targets: {},
+      labels: {},
+      showMore: false,
+    };
+  }
+
+  componentDidMount() {
+    fetch(this.props.pathPrefix + '/api/v1/targets')
+      .then(resp => {
+        if (resp.ok) {
+          return resp.json();
+        } else {
+          throw new Error('Unexpected response status when fetching services: ' + resp.statusText); // TODO extract error
+        }
+      })
+      .then(json => {
+        this.setState({ response: json, targets: this.processTargets(json), labels: this.processLabels(json) });
+      })
+      .catch(err => {
+        this.setState({ fetchServicesErrorMessage: err });
+      });
+  }
+
+  isHealthy = (health: string) => {
+    return health === 'up';
+  };
+
+  processTargets = (resp: any) => {
+    const activeTargets = resp.data.activeTargets,
+      targets: any = Object.create({});
```

Comment:
Any reason why not just:

```suggestion
      targets: any = {};
```